array_api_compat/common/_aliases.py

```diff
@@ -5,36 +5,44 @@
 from __future__ import annotations
 
 import inspect
-from typing import NamedTuple, Optional, Sequence, Tuple, Union
-
-from ._helpers import array_namespace, _check_device, device, is_cupy_namespace
+from typing import TYPE_CHECKING, NamedTuple, Optional, Sequence, cast
+
+from ._helpers import (
+    _check_device,  # pyright: ignore[reportPrivateUsage]
+    array_namespace,
+    device,
+    is_cupy_namespace,
+)
 from ._typing import Array, Device, DType, Namespace
 
+if TYPE_CHECKING:
+    from typing_extensions import TypeIs
```

Comment:
```suggestion
    # TODO: import from typing (requires Python >=3.13)
    from typing_extensions import TypeIs
```
these comments make upgrading minimum python version so much easier.